list_s3_objects.py

```diff
@@ -0,0 +1,38 @@
+import boto3
+import json
+import hashlib
+
+# Replace these values with your AWS credentials and region
+aws_access_key_id = 'YOUR_ACCESS_KEY_ID'
+aws_secret_access_key = 'YOUR_SECRET_ACCESS_KEY'
+region_name = 'ap-southeast-2'
+bucket_name = 'YOUR_BUCKET_NAME'
+queue_url = 'YOUR_QUEUE_URL'
+session = boto3.Session()
+s3_client = session.client('s3', aws_access_key_id=aws_access_key_id, aws_secret_access_key=aws_secret_access_key, region_name=region_name)
+sqs_client = session.client('sqs', aws_access_key_id=aws_access_key_id, aws_secret_access_key=aws_secret_access_key, region_name=region_name)
+
+def list_s3_objects(bucket):
+    response = s3_client.list_objects(Bucket=bucket_name)

```

Comment:
How would you expand this to deal with a bucket containing more than 1000 objects, thus requiring pagination?

Reply:
Pagination is used in boto3 client.  To parallelize the listing process, we could use asyncio and ProcessPoolExecutor to deal large number of objects.